### DISCUSSION TITLE
OSC对鼠标响应太灵敏

### DISCUSSION THREAD

**Discussion by Frozen1084** [hooke007/mpv_PlayKit#141] [mpv 通用]
图片是和同样隐藏边框的PotPlayer的对比。可见当鼠标快速穿过边框时，PotPlayer是不响应的，但又完全不影响正常响应。反观MPV_lazy则过于频繁地响应，造成闪烁。
https://i.imgur.com/Pg8CQ9w.gif

还有为何菜单响应这么迟钝呢？从右键弹起到显示菜单，几乎有一秒的延迟。有什么办法可以立即显示吗？

**Comment by hooke007** [comment]
OSC的显隐和边框没有关系。原版也没有延迟响应，我没改这方面。有想法的欢迎PR

因为是独立的程序所以菜单冷启动慢这个文档里有说过，如果二次启动菜单还是很慢。。。这不应该，我不知道原因

**Reply by Frozen1084** [reply]
我说的就是热显示菜单的延迟。刚试了下mpvnet-CM，几乎没有延迟。

**Reply by hooke007** [reply]
当然不会延迟。。。mpv的菜单是独立的程序，它接收到你的快捷键指令才开始启动。CM的菜单是UI的一部分，他在你启动程序的时候已经伴随启动了，这个说法好理解吧。

**Comment by Frozen1084** [comment]
谢谢。我理解。我知道它是独立的，我说的也不是第一次显示时的慢。

另外能禁止显示，用左右方向键调整进度时的提示信息吗？它太大，太显眼了，简直影响正常的视频观看了。
这要是放在调试模式上可以接受，日常使用则简单是一种干扰。
https://i.imgur.com/duTaDMY.gif

还有"回车"键是全屏，却不能从全屏切换出来。不合理。

**Reply by hooke007** [reply]
二次响应慢（一秒）无法复现。

我觉得都很合理。不喜欢改快捷键就是了。input.conf
`LEFT no-osd seek -5`

**Reply by Frozen1084** [reply]
我使用的是默认配置，用AutoHotkey测试计时，热显示大约300毫秒左右。没有我感觉的1秒，但也与mpvnet-CM有巨大的差别。这是正常的吗？
https://i.imgur.com/kgx5f8p.gif
```
f1::
{
	send "{RButton}"

	ti := A_TickCount
	loop
	{
		MouseGetPos , , &id
		t := wingetclass(id)
		if t = "#32768"
			break
		else

			sleep 30

	}
	MsgBox  A_TickCount - ti
}
```

**Reply by hooke007** [reply]
正常，我这里从触发到显示的第一帧差不多288ms

从脚本上讲，每次接收到指令，然后开始编译并绘制，这个间隔对我来讲已经够快了。
当然我不懂tcl的设计，只是单纯移植脚本到win上使用而已，没法优化性能

edit：测了CM这种原生框架的响应速度是144-216ms，所以我觉得能接受
